pybamm/expression_tree/unary_operators.py

```diff
@@ -34,6 +34,21 @@ def __init__(self, name, child, domains=None):
         super().__init__(name, children=[child], domains=domains)
         self.child = self.children[0]
 
+    @classmethod
+    def _from_json(cls, name, snippet: dict):
+        """Use to instantiate when deserialising"""
+
+        instance = cls.__new__(cls)
+
+        super(UnaryOperator, instance).__init__(
+            name,
```

Comment:
is `name` here going to be the same as `self.name`?

Reply:
It should be, yes - I'll edit this to be more in line with other similarly structured subclasses (e.g. in binary_operators.py), where the superclass _from_json() function just takes the dictionary snippet, taking the name from that dictionary.